### PR TITLE
remove internal openshift refs from oauthserver

### PR DESCRIPTION
**Pull request by deads2k**
Updates oauthserver tests to use clients for mocks instead of internal registry shims from the storage.

@openshift/sig-security

### CODE REVIEW

File: pkg/oauthserver/userregistry/identitymapper/lookup_test.go [hunk 1]
```diff
@@ -25,226 +17,76 @@ func TestLookup(t *testing.T) {
...
-			Actions:  &actions,
-		}
-		if tc.ExistingIdentity != nil {
-			identityRegistry.GetIdentities[tc.ExistingIdentity.Name] = tc.ExistingIdentity
-		}
-		if tc.ExistingUser != nil {
-			userRegistry.GetUsers[tc.ExistingUser.Name] = tc.ExistingUser
-		}
-
-		mappingStorage := mappingregistry.NewREST(userRegistry, identityRegistry)
```
Comment (deads2k) [inline pkg/oauthserver/userregistry/identitymapper/lookup_test.go]: this mocking structure was effectively re-running tests from here: https://github.com/openshift/origin/blob/master/pkg/user/registry/useridentitymapping/rest_test.go .  The code in this package relies on the external client.

File: pkg/oauthserver/userregistry/identitymapper/provision_test.go [hunk 2]
```diff
@@ -52,241 +57,295 @@ func TestProvision(t *testing.T) {
...
 			},
 
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
+			ValidateActions: func(t *testing.T, actions []clienttesting.Action) {
+				if len(actions) != 3 {
+					t.Fatal(spew.Sdump(actions))
+				}
+				if !actions[0].Matches("get", "identities") || actions[0].(clienttesting.GetAction).GetName() != "idp:bob" {
+					t.Error(spew.Sdump(actions))
```
Comment (simo5) [inline pkg/oauthserver/userregistry/identitymapper/provision_test.go]: Shouldn't this be t.Error(spew.Sdump(actions[0]))
(and similar below ?)

Comment (simo5) [reply pkg/oauthserver/userregistry/identitymapper/provision_test.go]: Or perhaps even add what action failed exactly to the error, maybe via helper like:
```
func checkAction(num int, actions []clienttesting.Action, match string[]) {
    if !actions[num].Matches(**match) {
        t.Errorf("Expected action %d to match %v, found %s", num, match, spew.Sdump(actions[num]))
    }
}
```
then:
```
    checkAction(actions, 0, ["create", "identities"])
```
and to check names:
```
func checkActionAndName(num int, actions []clienttesting.Action, match string[], name) {
    if !actions[num].Matches(**match) ||  actions[num].(clienttesting.GetAction).GetName() != name {
        t.Errorf("Expected action %d to match %v, found %s", num, match, spew.Sdump(actions[num]))
    }
}
```
then:
```
    checkActionAndName(actions, 0, ["get", "identities"], "idp:bob")
```
and so on...

File: pkg/oauthserver/userregistry/identitymapper/provision_test.go [hunk 3]
```diff
@@ -52,241 +57,295 @@ func TestProvision(t *testing.T) {
...
 			NewIdentityGetterResponses: []interface{}{
 				kerrs.NewAlreadyExists(userapi.Resource("User"), "bob"),
 				makeUser("bobUserUID", "bob", "idp:bob"),
 			},
 
-			ExpectedActions: []test.Action{
-				{Name: "GetIdentity", Object: "idp:bob"},
+			ValidateActions: func(t *testing.T, actions []clienttesting.Action) {
+				if len(actions) != 3 {
+					t.Fatal(spew.Sdump(actions))
```
Comment (simo5) [inline pkg/oauthserver/userregistry/identitymapper/provision_test.go]: And here:
```
t.Fatalf("Expected 3 actions, found %q", spew.Sdump(actions))
```
?

**Comment by simo5** [comment]
/lgtm
line numbers give enough context for tests

**Pull request merged**